File: EnvMonitorAndroid/app/src/main/java/com/envmonitor/models/EnvParameter.kt
package com.envmonitor.models

/**
 * Mô hình dữ liệu cho một thông số môi trường
 * 
 * @property value Giá trị của thông số
 * @property unit Đơn vị đo lường
 * @property status Trạng thái dựa trên ngưỡng (normal, warning, kém, danger)
 * @property timestamp Thời gian cập nhật (format HH:mm:ss)
 * @property lastUpdate Thời điểm cập nhật cuối cùng (timestamp Unix hoặc null)
 */
data class EnvParameter(
    val value: Double,
    val unit: String,
    val status: String,
    val timestamp: String,
    val lastUpdate: Long? = null
) {

    /**
     * Lấy văn bản trạng thái phù hợp dựa trên status
     */
    fun getStatusText(): String {
        return when (status) {
            "normal" -> "Tốt"
            "warning" -> "Trung bình"
            "kém" -> "Kém"
            "danger" -> "Xấu"
            "offline" -> "Offline"
            else -> "Không xác định"
        }
    }
    
    /**
     * Kiểm tra xem thông số có vượt quá ngưỡng cảnh báo hay không
     * @return true nếu thông số vượt quá ngưỡng cảnh báo
     */
    fun isWarningOrAbove(): Boolean {
        return status == "warning" || status == "kém" || status == "danger"
    }
    
    /**
     * Kiểm tra xem thông số có vượt quá ngưỡng nguy hiểm hay không
     * @return true nếu thông số vượt quá ngưỡng nguy hiểm
     */
    fun isDanger(): Boolean {
        return status == "danger"
    }
}

File: env_monitor_app/lib/models/env_parameter.dart
class EnvParameter {
  final double value;
  final String unit;
  final String status;
  final String timestamp;
  final int? lastUpdate;

  EnvParameter({
    required this.value,
    required this.unit,
    required this.status,
    required this.timestamp,
    this.lastUpdate,
  });

  factory EnvParameter.fromJson(Map<String, dynamic> json) {
    return EnvParameter(
      value: json['value'] is int 
          ? (json['value'] as int).toDouble() 
          : json['value'] as double,
      unit: json['unit'] as String,
      status: json['status'] as String,
      timestamp: json['timestamp'] as String,
      lastUpdate: json['last_update'] as int?,
    );
  }

  /// Lấy văn bản trạng thái hiển thị
  String getStatusText() {
    switch (status) {
      case 'normal':
        return 'Tốt';
      case 'warning':
        return 'Trung bình';
      case 'kém':
        return 'Kém';
      case 'danger':
        return 'Xấu';
      default:
        return 'Không xác định';
    }
  }

  /// Kiểm tra xem thông số có ở trạng thái nguy hiểm không
  bool isDanger() {
    return status == 'danger';
  }

  /// Kiểm tra xem thông số có ở trạng thái cảnh báo không
  bool isWarning() {
    return status == 'warning' || status == 'kém';
  }
}